### PR TITLE
Add git.RemoteTags() function

### PR DESCRIPTION
**Pull request by puerco**
#### What type of PR is this?
/kind feature

#### What this PR does / why we need it:

This PR adds a new function to the git package: `git.RemoteTags()` which queries the default remote repository and returns it's tags in a string slice.

It also modifies the `git.LsRemote()` function so that it honors `Repo.maxRetries`  when performing the remote query. 

#### Which issue(s) this PR fixes:
Related to #1595 

#### Special notes for your reviewer:

Depends on #1621 to run

#### Does this PR introduce a user-facing change?

```release-note
- New remote tag functions: `git.RemoteTags()` and `git.HasRemoteTag()` which return the tags in the default remote repository and check if one exists
- `git.LSRemote()` now honors `Repo.maxRetries` and will retry when a network failure occurs
```

### CODE REVIEW

File: pkg/git/git.go [hunk 1]
```diff
@@ -1031,6 +1048,44 @@ func (r *Repo) IsDirty() (bool, error) {
...
+// RemoteTags return the tags that currently exist in the
+func (r *Repo) RemoteTags() (tags []string, err error) {
+	logrus.Debug("Listing remote tags with ls-remote")
+	output, err := r.LsRemote("--tags", DefaultRemote)
+	if err != nil {
+		return tags, errors.Wrap(err, "while listing tags using ls-remote")
+	}
+	const tagPreRef = "refs/tags/"
+	tags = make([]string, 0)
+	for _, line := range strings.Split(output, "\n") {
```
Comment (saschagrunert) [inline pkg/git/git.go:1060]: I'd prefer using `bufio.Scanner` here.

File: pkg/git/git.go [hunk 2]
```diff
@@ -997,8 +997,25 @@ func (r *Repo) PushToRemote(remote, remoteBranch string) error {
...
+			return output, nil
+		}
+		err = NewNetworkError(err)
+		if !err.(NetworkError).CanRetry() || r.maxRetries == 0 || i == 1 {
+			return "", err
+		}
+
+		waitTime := math.Pow(2, float64(r.maxRetries-i))
+		logrus.Errorf(
+			"Error executing ls-remote (will retry %d more times in %.0f secs): %s",
```
Comment (saschagrunert) [inline pkg/git/git.go:1013]:
```suggestion
			"Executing ls-remote (will retry %d more times in %.0f secs): %s",
```
The log level already indicates that it's an error.

**Review by saschagrunert — APPROVED** [review]
/lgtm
/hold
for rebase

File: pkg/git/git.go (hunk 1 above)
Comment (puerco) [reply pkg/git/git.go]: Actually, I prefer it too. Changed :)

**Review by saschagrunert — APPROVED** [review]
/lgtm

**Pull request merged**